Assets/Scripts/Data Model/GameState.cs

```diff
@@ -1,191 +1,190 @@
-			}
-			else if (gameState == PlanningState.Play)
-			{
-				TimeManager.instance.PlayStateEntered();
-			}
-			else if (gameState == PlanningState.FastForward)
-			{
-				TimeManager.instance.FastForwardStateEntered();
-			}
-			else if (gameState == PlanningState.Simulation)
+			if (!firstUpdateComplete)
 			{
-				OnSimulationPhaseStarted();
-			}
-			else if (gameState == PlanningState.End)
-			{
-				InterfaceCanvas.Instance.menuBarPlanWizard.toggle.interactable = false;
+				firstUpdateComplete = true;
+				PlanningState tempState = StringToPlanningState(state.state); 
+				if(tempState != PlanningState.Setup)
+					TimeManager.instance.SetupStateExited();
 			}
 
-			//Old state left
-			if (prevState == PlanningState.Simulation)
+			int prevMonth = month;
+			PlanningState prevState = gameState;
+
+			gameState = StringToPlanningState(state.state);
+			month = Util.ParseToInt(state.month, 0);
+
+			//Month change
+			if (month != prevMonth)
 			{
-				OnSimulationPhaseEnded();
+				if (gameState != PlanningState.Setup)
+					PlanManager.MonthTick(month);
+				PlanWizard.UpdateMinSelectableTime();
+
+				if (OnCurrentMonthChanged != null)
+				{
+					OnCurrentMonthChanged.Invoke(prevMonth, month);
+				}
 			}
-			else if (prevState == PlanningState.Setup)
+
+			//State change
+			if (gameState != prevState)
 			{
-				OnSetupPhaseEnded();
+				EraHUD.instance.State = gameState;
+				//New state entered
+				if (gameState == PlanningState.Setup)
+				{
+					TimeManager.instance.SetupStateEntered();
+					OnSetupPhaseStarted();
+				}
+				else if (gameState == PlanningState.Pause)
+				{
+					TimeManager.instance.PauseStateEntered();
+				}
+				else if (gameState == PlanningState.Play)
+				{
+					TimeManager.instance.PlayStateEntered();
+				}
+				else if (gameState == PlanningState.FastForward)
+				{
+					TimeManager.instance.FastForwardStateEntered();
+				}
+				else if (gameState == PlanningState.Simulation)
+				{
+					OnSimulationPhaseStarted();
+				}
+				else if (gameState == PlanningState.End)
+				{
+					InterfaceCanvas.Instance.menuBarPlanWizard.toggle.interactable = false;
+				}
+
+				//Old state left
+				if (prevState == PlanningState.Simulation)
+				{
+					OnSimulationPhaseEnded();
+				}
+				else if (prevState == PlanningState.Setup)
+				{
+					OnSetupPhaseEnded();
+				}
 			}
+			//Update UI
+			timeBar.SetDate(month);
+			TimeManager.instance.UpdateUI(state);
 		}
-		//Update UI
-		timeBar.SetDate(month);
-		TimeManager.instance.UpdateUI(state);
-	}
 
-	private static void OnSetupPhaseStarted()
-	{
-		if(!TeamManager.AreWeGameMaster)
-			InterfaceCanvas.Instance.menuBarPlanWizard.toggle.interactable = false;
-		//PlansMonitor.RefreshPlanEditButtonInteractablity();
-	}
+		private static void OnSetupPhaseStarted()
+		{
+			if(!TeamManager.AreWeGameMaster)
+				InterfaceCanvas.Instance.menuBarPlanWizard.toggle.interactable = false;
+			//PlansMonitor.RefreshPlanEditButtonInteractablity();
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)